src/Objs/Redirect/RedirectComponent.tsx

```diff
@@ -1,19 +1,20 @@
 import * as React from 'react'
-import * as Scrivito from 'scrivito'
+import { load, provideComponent, urlFor } from 'scrivito'
 import { InPlaceEditingPlaceholder } from '../../Components/InPlaceEditingPlaceholder'
+import { Redirect } from './RedirectObjClass'
 
-Scrivito.provideComponent('Redirect', ({ page }) => {
+provideComponent(Redirect, ({ page }) => {
   React.useEffect(() => {
-    Scrivito.load(() => {
+    load(() => {
       const link = page.get('link')
-      const url = link && Scrivito.urlFor(link)
+      const url = link && urlFor(link)
 
       return { link, url }
     }).then(({ link, url }) => {
-      if (!link) return
+      if (!link || !url) return
 
       if (link.isExternal()) {
-        window.top.location.replace(url)
+        window.top?.location.replace(url)
```

Comment:
* [ ] Not sure under which circumstances `window.top` is `null`. Maybe falling back to the `else` block is the better strategy?

Reply:
I cleaned it up for good order.

FYI this is what Copilot has to say (not checked):

> In a typical browser environment, window.top should never be null because even if the window isn't part of a frameset, window.top will simply return a reference to the window itself.
> 
> However, TypeScript is designed to be safe and assumes that any property of the window object could potentially be null. This is because TypeScript can be used in a variety of environments, not just browsers, and in some of these environments, the window object or its properties might not be available.
> 
> In practice, if you're running your code in a browser environment and you're not doing anything unusual with the window object, you can safely assume that window.top will never be null.